src/borg/testsuite/archiver.py

```diff
@@ -2023,17 +2024,108 @@ def test_prune_repository(self):
+        self.cmd('init', '--encryption=repokey', self.repository_location)
+        # Archives that will be kept, per the example
+        # Oldest archive
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 1, 1, 0, 0, 0), self.repository_location + '::test01', src_dir)
+        # 6 monthly archives
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 6, 30, 0, 0, 0), self.repository_location + '::test02', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 7, 31, 0, 0, 0), self.repository_location + '::test03', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 8, 31, 0, 0, 0), self.repository_location + '::test04', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 9, 30, 0, 0, 0), self.repository_location + '::test05', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 10, 31, 0, 0, 0), self.repository_location + '::test06', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 11, 30, 0, 0, 0), self.repository_location + '::test07', src_dir)
+        # 14 daily archives
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 17, 0, 0, 0), self.repository_location + '::test08', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 18, 0, 0, 0), self.repository_location + '::test09', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 20, 0, 0, 0), self.repository_location + '::test10', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 21, 0, 0, 0), self.repository_location + '::test11', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 22, 0, 0, 0), self.repository_location + '::test12', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 23, 0, 0, 0), self.repository_location + '::test13', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 24, 0, 0, 0), self.repository_location + '::test14', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 25, 0, 0, 0), self.repository_location + '::test15', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 26, 0, 0, 0), self.repository_location + '::test16', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 27, 0, 0, 0), self.repository_location + '::test17', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 28, 0, 0, 0), self.repository_location + '::test18', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 29, 0, 0, 0), self.repository_location + '::test19', src_dir)
+        self.cmd('create', '--timestamp', self._to_utc_timestamp(2015, 12, 30, 0, 0, 0), self.repository_location + '::test20', src_dir)
```

Comment:
idea how to make this code easier / less duplicate:

```
def create_archive_ts(name, y, m, d, H=0, M=0, S=0):
    loc = self.repository_location + '::' + name
    self.cmd('create', '--timestamp', self._to_utc_timestamp(y, m, d, H, M, S), loc, src_dir)
```

... and then just call that.

Reply:
thanks for fixing!